clients/buckets/bucket.go

```diff
@@ -161,6 +165,18 @@ func (c Client) List(ctx context.Context) (ListResponse, error) {
 	if err != nil {
 		return ListResponse{}, fmt.Errorf("failed to list buckets:%w", err)
 	}
+
+	// if the response has code != 2xx return empty list with response info
+	if !rest.IsSuccess(resp) {
+		body, err := io.ReadAll(resp.Body)
+		resp.Body.Close()
+		if err != nil {
+			return ListResponse{}, err
```

Comment:
You're swallowing the actual response error in case we can't read the body. This will make it impossible for a user of the client to actually know that and how the REST API call has failed. I'd still return an API error without body and log that we failed to read the body

Reply:
Actually this is something that's now happening in several cases in this client. @warber is this also a pattern you've adopted in the other client changes? IMO that's problematic when using these clients. Reading the response body is not the main action of these methods, so we shouldn't just not return any information about the actual REST response if that fails.

Reply:
That is very true: the original error is more important - I will update all of these 👍

Reply:
@warber, we should align all clients here